primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -252,8 +253,24 @@ private static boolean isValidFileContent(PrimeApplicationContext context, FileU
                 }
             }
 
+
             String contentType = context.getFileTypeDetector().probeContentType(tempFile);
 
+            // We want to force the FileTypeDetector impl to try to detect the file type by its content,
+            // this is why we intentionally hide the original file name/extension at first because both
+            // tika and mime-types go the easy way of looking up the file extension if possible.
+            // If this first attempt failed we try again, but now while preserving the original file name/extension
+            // to e.g. make the JDK default FileTypeDetector work
+            if (contentType == null) {
+                String fileExtension = getFileExtension(fileName);
+
+                if (fileExtension != null) {
+                    Path newTempFile = Files.createTempFile(tempFilePrefix, fileExtension);
+                    tempFile = Files.move(tempFile, newTempFile, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Can you try renaming the file without creating a new temp file. I found this:

```java
Path source = Paths.get("Notes.txt");
Files.move(source, source.resolveSibling("Keynotes.txt"),
            StandardCopyOption.REPLACE_EXISTING);
```

Source: https://www.codejava.net/java-se/file-io/how-to-rename-move-file-or-directory-in-java

Reply:
done